### PR TITLE
feat: add support for viml

### PR DESCRIPTION
**Pull request by vigoux**
As asked in #1187 this PR will be tracking the progress in supporting
VimL.

As the parser is not yet finished, this is really just a draft branch,
so that people can subscribe to it and have updates on the progress.

### CODE REVIEW

File: queries/vim/injections.scm [hunk 1]
```diff
@@ -0,0 +1,6 @@
+(lua_statement . (_) @lua)
+(ruby_statement . (_) @ruby)
+(python_statement . (_) @python)
```
Comment (stsewd) [inline queries/vim/injections.scm:3]:
```suggestion
(python_statement . (_) @python)
(comment) @comment
```

File: queries/vim/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+] @repeat
+
+[
+  "function"
+  "endfunction"
+] @keyword.function
+
+[ (bang) (spread) ] @punctuation.special
+
+[ (no_option) (inv_option) (default_option) (option_name) ] @constant.builtin
```
Comment (theHamsta) [inline queries/vim/highlights.scm:33]: are options constant when you can set them? `@variable.builtin`?

Comment (vigoux) [reply queries/vim/highlights.scm:33]: Huh, yup, that seems better.

**Comment by vigoux** [comment]
After quite some time, I think this is pretty much ready.

**Comment by shadmansaleh** [comment]
@vigoux I think this is this a bug.
Screenshot

```
" .. a
```
is highlighted as comment . While this " is end of string

Also if I comment out the let statement . It recognises the string correctly but endf looses highlight.
Screenshot

and endf is highlighted again if I change a to ab 😕
Screenshot_20210812-190458_Termux

Can you reproduce this ?

**Comment by vigoux** [comment]
@shadmansaleh can reproduce in the parser too, let me check what the problem is.

**Comment by vigoux** [comment]
Found a fix, pushing in few minutes.

**Comment by vigoux** [comment]
And pushed !  (sorry for the noise)

**Comment by shadmansaleh** [comment]
> And pushed ! (sorry for the noise)

Works now confirmed. Thanks for the quick fix . I'll open issue in the parser instead of commenting here if I see something funny. I'm Sorry too :)

**Comment by vigoux** [comment]
Don't worry about that, and keep those bug reports coming, I need it !

**Comment by vigoux** [comment]
I am in the process of adding things, so please report any missing highlight

**Comment by stsewd** [comment]
The windows CI is broken since this change, not sure why it fails after installing some parsers (there are like 3 PRs with the same problem)

**Comment by jdhao** [comment]
@vigoux  Some highlights are missing strangely and `"` in mapping is treated as comments. See image below:

image

**Comment by vigoux** [comment]
Hey, this is a known, bug that has been reported into `tree-sitter-viml` please report bugs there.

**Pull request merged**